lib/mix/lib/mix/tasks/compile.app.ex

```diff
@@ -199,73 +208,122 @@ defmodule Mix.Tasks.Compile.App do
+
       {:included_applications, value} ->
         unless is_list(value) and Enum.all?(value, &is_atom(&1)) do
-          Mix.raise "Application included applications (:included_applications) should be a list of atoms, got: #{inspect value}"
+          Mix.raise(
+            "Application included applications (:included_applications) should be a list of atoms, got: #{
+              inspect(value)
+            }"
+          )
         end
+
       {:extra_applications, value} ->
         unless is_list(value) and Enum.all?(value, &is_atom(&1)) do
-          Mix.raise "Application extra applications (:extra_applications) should be a list of atoms, got: #{inspect value}"
+          Mix.raise(
+            "Application extra applications (:extra_applications) should be a list of atoms, got: #{
+              inspect(value)
+            }"
+          )
         end
+
       {:applications, value} ->
         unless is_list(value) and Enum.all?(value, &is_atom(&1)) do
-          Mix.raise "Application applications (:applications) should be a list of atoms, got: #{inspect value}"
+          Mix.raise(
+            "Application applications (:applications) should be a list of atoms, got: #{
+              inspect(value)
+            }"
+          )
         end
+
       {:env, value} ->
         unless Keyword.keyword?(value) do
-          Mix.raise "Application environment (:env) should be a keyword list, got: #{inspect value}"
+          Mix.raise(
+            "Application environment (:env) should be a keyword list, got: #{inspect(value)}"
+          )
         end
+
       {:start_phases, value} ->
         unless Keyword.keyword?(value) do
-          Mix.raise "Application start phases (:start_phases) should be a keyword list, got: #{inspect value}"
+          Mix.raise(
+            "Application start phases (:start_phases) should be a keyword list, got: #{
+              inspect(value)
+            }"
+          )
         end
+
       {:mod, []} ->
         :ok
+
       {:mod, {module, _args}} when is_atom(module) ->
         :ok
+
       {:mod, value} ->
-        Mix.raise "Application callback module (:mod) should be either [] or {module, start_args}, got: #{inspect value}"
+        Mix.raise(
+          "Application callback module (:mod) should be either [] or {module, start_args}, got: #{
+            inspect(value)
+          }"
+        )
+
       _ ->
         :ok
-    end
+    end)
 
     properties
   end
 
   defp apps_from_prod_non_optional_deps(properties) do
     included_applications = Keyword.get(properties, :included_applications, [])
 
-    for %{app: app, opts: opts, top_level: true} <- Mix.Dep.cached,
+    for %{app: app, opts: opts, top_level: true} <- Mix.Dep.cached(),
         Keyword.get(opts, :app, true),
         Keyword.get(opts, :runtime, true),
         not Keyword.get(opts, :optional, false),
-        app not in included_applications,
-        do: app
+        app not in included_applications, do: app
```

Comment:
I'm not sure about location `do` block on one line with condition

Reply:
Good catch. This is a bug. I will investigate!